pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_scheduling.go

```diff
@@ -193,6 +261,92 @@ func (r *schedulingReconciler) chooseShardAndMarkCondition(logger klog.Logger, w
 	return targetShard.Name, nil
 }
 
-func isValidShard(shard *tenancyv1alpha1.ClusterWorkspaceShard) (valid bool, reason, message string) {
+func (r *schedulingReconciler) createThisWorkspace(ctx context.Context, cluster logicalcluster.Name, shard *tenancyv1alpha1.ClusterWorkspaceShard, workspace *tenancyv1alpha1.ClusterWorkspace) error {
+	this := &tenancyv1alpha1.ThisWorkspace{
+		// TODO(p0lyn0mial): in the future we could set an UID based back-reference to ClusterWorkspace as annotation
+		ObjectMeta: metav1.ObjectMeta{Name: "this"},
+		// TODO: set Type on the spec ?!
+		Spec: tenancyv1alpha1.ThisWorkspaceSpec{},
+	}
+	logicalClusterAdminClient, err := r.kcpLogicalClusterAdminClientFor(shard)
+	if err != nil {
+		return err
+	}
+	_, err = logicalClusterAdminClient.Cluster(cluster).TenancyV1alpha1().ThisWorkspaces().Create(ctx, this, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
this contradicts the function name. Would it up into the main func (I commented). Makes it clearer.